IBM-Cloud/ibm-developer-extension-vscode
# feat: Added acct, region, api, resource commands and improved ux for deploy

# Context

This PR will add a few new VSCode commands as well as improve on UX for deploy. The following commands will be added:

- `ibmcloud account list`
- `ibmcloud account show`
- `ibmcloud account users`
- `ibmcloud api`
- `ibmcloud regions`
- `ibmcloud target`
- `ibmcloud resource service-instances`

In addition, the README has been updated to removed the old YouTube links and clarified some of the dev flow instructions.


# Breaking Change
- The `ibmcloud dev deploy` command will only allow deploying to IBM Cloud Kubernetes Service


# Steps To Test

## Setup
1. Install `vsce` npm package if not already: `npm install -g vsce`
2. Checkout PR branch
3. Run the command `vsce package`
4. Verify vsix package is built successfully
5. Open a starterkit project in VSCode
6. Open the `.ibm-project` file and remove the values for the `image-deploy-target` and `cluster-name` if values exist
    - The `dev deploy` command will use the values in these properties to determine what cluster and image target to deploy. If they are not valid the deployment will fail
7. Go to **Extension -> Install From VSIX`** and select vsix file created in step 4
![2022-04-01_15-18](https://user-images.githubusercontent.com/5360960/161335625-a380db81-a8ff-4b7d-8567-d6e62d3b715e.png)
8. Reload VSCode when requested

## Test new commands
**Note:** 

1. Make sure to follow the steps in the **Setup** section
2. Open the command palette 
  - To enter Command Palette press `Ctrl+Shift+P` or `F1` or navigate to `View-> Command Palette`
3. Enter `ibmcloud login --sso`
4. Follow the instructions to login into IBM Cloud
5. Repeat step 2 (command palette) 
6. Enter the command `ibmcloud account list`
8. Verify that step 6 runs successfully
9. Repeat steps 5-7 except use the following commands: 
    - `ibmcloud account show`
    - `ibmcloud account users`
    - `ibmcloud api`
    - `ibmcloud regions`
    - `ibmcloud target`
    - `ibmcloud resource service-instances`


## Test new UX for dev deploy
**Note:** Before proceeding make sure that you have a working IBM Kubernetes Service (IKS) cluster and an image namespace in IBM Container Registry (CR)

1. Make sure to follow the steps in the **Setup** section
2. Login into to IBM Cloud by entering the command `ibmcloud login --sso` in the Command Palette
3. Verify login is successful
4. Build the project by entering the command `ibmcloud dev build (release)` in the Command Palette
5. Verify build is successful
6. Start deployment by entering the command `ibmcloud dev deploy` in the Command Palette
7. Verify that terminal window is opened with the `ibmcloud dev deploy` executed
8. Follow the prompts provided and verify that app is deployed successfully.
13. Click on the link at the bottom of the terminal output and verify that app can be viewed in the browser


# Videos
![new_dev_deploy_iks](https://user-images.githubusercontent.com/5360960/165140949-16e34e49-6eb7-41a1-b10e-0b892d1b4734.gif)
